### PR TITLE
fix(tianmu):  the sock file directory of ./reinstall.sh is inconsistent with the directory in my.cnf(#1075)

### LINKED ISSUE #1075: bug: ./reinstall.sh automatic installation, the sock file directory is inconsistent with the directory in my.cnf
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

the sock file directory in my.cnf which generated by default as follows
`socket = /tmp/mysql.sock`
```shell
[client]
port = 3306
socket          = /tmp/mysql.sock

[mysqld]
port                            = 3306
basedir                         = /stonedb57/install/
character-sets-dir              = /stonedb57/install/share/charsets/
plugin_dir                      = /stonedb57/install/lib/plugin/
tmpdir                          = /stonedb57/install/tmp/
socket                          = /tmp/mysql.sock
datadir                         = /stonedb57/install/data/
pid-file                        = /stonedb57/install/data/mysqld.pid
log-error                       = /stonedb57/install/data/mysqld.log
lc-messages-dir                 = /stonedb57/install/share/
lc_messages                     = en_US
```
But when using this script `./reinstall.sh` for automated installation, the sock file directory is `/stonedb57/install//tmp/mysql.sock`，the sock file directory is inconsistent with the directory in my.cnf.

as follows

```shell
+ echo mysqld_pid::: /stonedb57/install/data/mysqld.pid
mysqld_pid::: /stonedb57/install/data/mysqld.pid
+ wait_for_ready
+ i=0+
+ test 0 -ne/stonedb57/install//bin/mysqld_safe 900 --user=mysql
 --datadir=/stonedb57/install/data/ --socket=/stonedb57/install//tmp/mysql.sock --pid-file=/stonedb57/install/data/mysqld.
+ /stonedb57/install//bin/mysqladmin ping --socket=/stonedb57/install//tmp/mysql.sock
+ kill -0 126609
+ :
+ echo .\c
.+ expr 0 + 1
+ i=1
+ sleep 1
Logging to '/stonedb57/install/data/ub01.err'.
2022-12-06T08:15:56.815017Z mysqld_safe Starting mysqld daemon with databases from /stonedb57/install/data
+ test 1 -ne 900
+ /stonedb57/install//bin/mysqladmin ping --socket=/stonedb57/install//tmp/mysql.sock
```

### Expected behavior

The two directories should be consistent

### How To Reproduce

_No response_

### Environment

```shell
root@ub01:/stonedb57/install/bin# ./mysqld --version
./mysqld  Ver 5.7.36-StoneDB for Linux on x86_64 (build-)
build information as follow:
        Repository address: https://github.com/stoneatom/stonedb.git:stonedb-5.7-dev
        Branch name: stonedb-5.7-dev
        Last commit ID: e395f89e1
        Last commit time: Date:   Tue Dec 6 01:04:24 2022 +0800
        Build time: Date: Tue Dec  6 16:03:58 CST 2022
```

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

### PR DESCRIPTION
**Pull request by isredstar**
## Summary about this PR

Issue Number: close #1075 
**anlysis：** `mysql.sock` generated in the installation directory can resolve the inconsistency and avoid conflict with the mysql server in the environment.
**main fix:** `socket          = /tmp/mysql.sock` -> `socket          = YOUR_ABS_PATH/tmp/mysql.sock`

## Tests Check List

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

File: storage/tianmu/loader/load_parser.cpp [hunk 1]
```diff
@@ -175,15 +175,15 @@ bool LoadParser::MakeValue(uint att, ValueCache &buffer) {
...
 
-    if (value == 0)  // Value of auto inc column was not assigned by user
+    if (*buf == 0)  // Value of auto inc column was not assigned by user
       *buf = attr->AutoIncNext();
 
-    if (value > attr->GetAutoInc())
-      attr->SetAutoInc(value);
-
+    if (static_cast<uint64_t>(*buf) > attr->GetAutoInc()) {
+      if (*buf > 0 || ((attr->TypeName() == common::ColumnType::BIGINT) && attr->GetIfUnsigned()))
```
Comment (RingsC) [inline storage/tianmu/loader/load_parser.cpp]: Whether `*buf >0` statement can be move back to `if (*buf ==0)` statement to test the condition as soon as possible, and therefore, at this line we can only do typename and unsigned flag  check.

File: storage/tianmu/core/tianmu_table.cpp [hunk 2]
```diff
@@ -604,25 +604,26 @@ void TianmuTable::Field2VC(Field *f, loader::ValueCache &vc, size_t col) {
...
         *reinterpret_cast<int64_t *>(vc.Prepare(sizeof(int64_t))) = m_attrs[col]->AutoIncNext();
       else
-        *reinterpret_cast<int64_t *>(vc.Prepare(sizeof(int64_t))) = v;
+        *reinterpret_cast<int64_t *>(vc.Prepare(sizeof(int64_t))) = value;
       vc.ExpectedSize(sizeof(int64_t));
       if (m_attrs[col]->GetIfAutoInc()) {
         // inc counter should be set to value of user assigned
-        if (static_cast<uint64_t>(v) > m_attrs[col]->GetAutoInc()) {
-          m_attrs[col]->SetAutoInc(v);
+        if (static_cast<uint64_t>(value) > m_attrs[col]->GetAutoInc()) {
```
Comment (RingsC) [inline storage/tianmu/core/tianmu_table.cpp]: and how about this case:
`value` is less than `0`.

Comment (isredstar) [reply storage/tianmu/core/tianmu_table.cpp]: When `col value` is negative or greater than  INT64_MAX，`value `is less than 0.

Comment (isredstar) [reply storage/tianmu/core/tianmu_table.cpp]: In addition, `col_value < 0`  will not change  auto_increment value, which is  consistent with innodb

File: storage/tianmu/loader/load_parser.cpp (hunk 1 above)
Comment (isredstar) [reply storage/tianmu/loader/load_parser.cpp]: `*buf ==0` will acquire the auto_increment value, And ` *buf > 0` is in order to assign the next start value to the auto_increment. In addition,` *buf < 0` will not change auto_increment value, which is consistent with innodb，except unsigned bigint type which is greater than `INT64_MAX`.
When col_value  is greater than `INT64_MAX`, `*buf `will is negative.

Comment (isredstar) [reply storage/tianmu/loader/load_parser.cpp]: For example:
`insert into auto_table values(0, "nnn");  # *buf == 0`
`insert into auto_value values(1,"sss"); # * buf > 0`
`insert into auto_value values(-122,"sss"); # * buf < 0`
`insert into auto_value values(9223372036854775809, "dddd"); # 9223372036854775809> INT64_MAX, *buf <0`

**Review by RingsC — APPROVED** [review]

**Review by lujiashun — APPROVED** [review]
LGTM

**Pull request closed without merging**